### PR TITLE
Cards

### PR DESCRIPTION
**Pull request by patriciakasapa**

### CODE REVIEW

**Comment by sammoorhouse** [comment]
I think Heroku review apps will be useful for reviewing UI changes: https://devcenter.heroku.com/articles/github-integration-review-apps

**Comment by patriciakasapa** [comment]
https://turntabl-dev-cards-yroaeklcq7n.herokuapp.com/

File: server.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+const app = express();
+
+// Serve only the static files form the dist directory
+app.use(express.static(__dirname + "/dist/turntabl-developer-access-control"));
+
+app.use(cookieParser());
+app.use(
+  cookieSession({
+    name: "session",
+    keys: ["super secret"],
```
Comment (sammoorhouse) [inline server.js]: can we do better than this please

File: server.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+app.use(passport.session());
+
+passport.use(
+  new SamlStrategy(
+    {
+      protocol: "https://",
+      entryPoint: process.env.ENTRY_POINT, // SSO URL (Step 2)
+      issuer: process.env.ISSUER, // Entity ID (Step 4)
+      path: "/auth/saml/callback" // or callback is the ACS URL path (Step 4)
+      // cert: process.env.CERTIFICATE
```
Comment (sammoorhouse) [inline server.js]: what do these comments mean?

File: server.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    },
+    function(profile, done) {
+      // Parse user profile data
+      console.log("profile", profile);
```
Comment (sammoorhouse) [inline server.js]: can you remove this console log?

File: server.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
+const express = require("express");
+const path = require("path");
+const SamlStrategy = require("passport-saml").Strategy;
+const passport = require("passport");
+const bodyParser = require("body-parser");
+const cookieSession = require("cookie-session");
+const cookieParser = require("cookie-parser");
+let userEmail = "";
```
Comment (sammoorhouse) [inline server.js]: this should not be a server-side global

File: server.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  })
+);
+app.use(bodyParser.json());
+app.use(passport.initialize());
+app.use(passport.session());
+
+passport.use(
+  new SamlStrategy(
+    {
+      protocol: "https://",
```
Comment (sammoorhouse) [inline server.js:32]: remove these comments and add your own

File: server.js (hunk 2 above)
Comment (sammoorhouse) [inline server.js]: what does this mean and should we delete it or enable it?

File: server.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+      entryPoint: process.env.ENTRY_POINT, // SSO URL (Step 2)
+      issuer: process.env.ISSUER, // Entity ID (Step 4)
+      path: "/auth/saml/callback" // or callback is the ACS URL path (Step 4)
+      // cert: process.env.CERTIFICATE
+    },
+    function(profile, done) {
+      // Parse user profile data
```
Comment (sammoorhouse) [inline server.js]: this comment doesn't match the code

File: server.js (hunk 3 above)
Comment (sammoorhouse) [inline server.js]: remove these console logs

File: server.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+      console.log("profile", profile);
+
+      // console.log("assertion", profile.getAssertion.toString());
```
Comment (sammoorhouse) [inline server.js]: remove commented code

File: server.js (hunk 1 above)
Comment (sammoorhouse) [inline server.js]: what is "super secret" and why are we checking it in to github?

**Review by sammoorhouse — CHANGES_REQUESTED** [review]

File: server.js [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+// To serve just the static files form the dist directory
+app.use(express.static(__dirname + "/dist/turntabl-developer-access-control"));
+
+app.use(cookieParser());
+
+app.use(
+  // To protect the cookies
+  cookieSession({
+    name: "session",
+    keys: [process.env.SECRET],
```
Comment (sammoorhouse) [inline server.js:20]: Nice

File: server.js [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+app.use(bodyParser.json());
+app.use(passport.initialize());
+app.use(passport.session());
+
+passport.use(
+  new SamlStrategy(
+    {
+      protocol: "https://",
+      // SSO URL to load the server.js to get the various module exports after authentication.
```
Comment (sammoorhouse) [inline server.js:33]: Not really sure what this means, it doesn’t have much to do with modules...

**Review by sammoorhouse — APPROVED** [review]

**Pull request merged**